stdlib/3/unittest/__init__.pyi

```diff
@@ -92,51 +88,56 @@ class TestCase:
                           msg: Any = ...) -> _AssertRaisesContext[_E]: ...
     @overload
     def assertWarns(self,  # type: ignore
-                    exception: Union[Type[Warning], Tuple[Type[Warning], ...]],
+                    expected_warning: Union[Type[Warning], Tuple[Type[Warning], ...]],
                     callable: Callable[..., Any],
                     *args: Any, **kwargs: Any) -> None: ...
     @overload
     def assertWarns(self,
-                    exception: Union[Type[Warning], Tuple[Type[Warning], ...]],
+                    expected_warning: Union[Type[Warning], Tuple[Type[Warning], ...]],
                     msg: Any = ...) -> _AssertWarnsContext: ...
     @overload
     def assertWarnsRegex(self,  # type: ignore
-                         exception: Union[Type[Warning], Tuple[Type[Warning], ...]],
+                         expected_warning: Union[Type[Warning], Tuple[Type[Warning], ...]],
+                         expected_regex: Union[str, bytes, Pattern[str], Pattern[bytes]],
                          callable: Callable[..., Any],
                          *args: Any, **kwargs: Any) -> None: ...
     @overload
     def assertWarnsRegex(self,
-                         exception: Union[Type[Warning], Tuple[Type[Warning], ...]],
+                         expected_warning: Union[Type[Warning], Tuple[Type[Warning], ...]],
+                         expected_regex: Union[str, bytes, Pattern[str], Pattern[bytes]],
                          msg: Any = ...) -> _AssertWarnsContext: ...
     def assertLogs(
         self, logger: Optional[logging.Logger] = ...,
         level: Union[int, str, None] = ...
     ) -> _AssertLogsContext: ...
     def assertAlmostEqual(self, first: float, second: float, places: int = ...,
                           msg: Any = ..., delta: float = ...) -> None: ...
+    @overload
+    def assertNotAlmostEqual(self, first: float, second: float,
```

Comment:
Not sure about this one.

```python3
if delta is not None and places is not None:
    raise TypeError("specify delta or places not both")
```
https://github.com/python/cpython/blob/0f221d09cad46bee38d1b7a7822772df66c53028/Lib/unittest/case.py#L946-L947

Thats the reasoning, but not sure if typeshed should take into count things like this, let me know.

Reply:
Oh, I see. I think the current types aren't quite correct because delta is before msg on line 120 below, when in reality it should be after. It may be possible to get correct types if we use three overloads: one with neither place nor delta, one with only place, one with only delta. However, I'm also fine with just having a single entry in the stub.